frontend/src/components/GifWidget/GifWidget.tsx

```diff
@@ -0,0 +1,84 @@
+import { SetStateAction, useEffect, useState } from "react";
+import {
+    GifWidgetStyled,
+    GifWidgetContent,
+    GifWidgetSearch,
+    GifList,
+    GifListImage,
+    GifWidgetAtrtibutionSection,
+    GiphyAttributionLogo
+} from "./GifWidget.styled";
+import axios from "axios";
+import GiphyAttributionLogoImage from "assets/GiphyAttributionLogo.png";
+
+const GifWidget = () => {
+    const [gifsLoading, setGifsLoading] = useState(true);
+    const [gifList, setGifList] = useState<any[]>([]);
+    const [gifSearchInput, setGifSearchInput] = useState("");
+
+    const findGif = (e: { target: { value: SetStateAction<string> } }) => {
+        setGifSearchInput(e.target.value);
+        setGifsLoading(true);
+    };
+
+    const getGifUrl = (fullGifUrl: string) => {
+        // image url which should be passed into chat box
+        console.log("image url", fullGifUrl);
+    };
+
+    const renderGifs = () => (
+        <GifWidgetContent>
+            <GifWidgetSearch
+                type="text"
+                value={gifSearchInput}
+                onChange={findGif}
+                placeholder="find gif"
+            />
+            <GifList>
+                {!gifsLoading &&
+                    gifList.map((item, index) => (
```

Comment:
In this project we follow, the role, to create mapping in const inside component hook

Reply:
comment unclear.

Reply:
create it as a "const" arrow function, and use that helper function here, over that mapping inside html

Reply:
done